src/data/nat/digits.lean

```diff
@@ -471,6 +472,22 @@ begin
   exact base_pow_length_digits_le' b m,
 end
 
+/-! ### Binary -/
+lemma base_two_digits_eq_bits (n : ℕ) : digits 2 n = n.bits.map (λ b, cond b 1 0) :=
+begin
+  induction n using nat.binary_rec_from_one with b n h ih,
+  { simp, }, { simp, },
+  rw bits_append_bit, cases b,
+  { rw digits_def', simpa [nat.bit, nat.bit0_val n],
+    { refl, }, { simpa [pos_iff_ne_zero, bit_eq_zero_iff], }, },
+  { simp only [nat.bit, nat.bit1_val n, cond],
+    rw [add_comm, digits_add 2],
+    simpa,
+    { refl, }, { norm_num, }, { norm_num, }, },
+  intro, contradiction,
+end
```

Comment:
Since all the extra goals created by `rw` are quite short, I'd just inline them:
```suggestion
begin
  induction n using nat.binary_rec_from_one with b n h ih,
  { simp },
  { simp },
  rw bits_append_bit _ _ (λ hn, absurd hn h),
  cases b,
  { rw digits_def' (le_refl 2),
    { simpa [nat.bit, nat.bit0_val n] },
    { simpa [pos_iff_ne_zero, bit_eq_zero_iff] } },
  { simpa [nat.bit, nat.bit1_val n, add_comm, digits_add 2 le_rfl 1 n (by norm_num) (by norm_num)] }
end
```